### ISSUE TITLE
cannot reproduce from pre-trained model on martket 1501 stride 1

### ISSUE THREAD

**Issue by Anurag14** [huanghoujing/person-reid-triplet-loss-baseline#38]
Respected Author, 
I am getting a very weird result from loading your pre-trained model and trying to run the evaluation, gave the following command as described in your readme. 
```
python script/experiment/train.py \
-d '(0,)' \
--only_test true \
--dataset market1501 \
--last_conv_stride 1 \
--normalize_feature false \
--exp_dir ~/results \
--model_weight_file ~/weights/market1501_stride1/model_weight.pth
```
I am getting map of 2.42 and CMC rank1 of 11. 
Any pointers would really be appreciated. Thanks in advance!

**Comment by huanghoujing** [comment]
Hi, thanks for your attention. Make sure that Python and Pytorch versions are the same as in README.

**Comment by Anurag14** [comment]
Hi, thanks for your response, I am running on exact environment python 2.7 and torch 0.3 but I can't reproduce the results of pretrained model. I again downloaded the weights also checked for my sci-kit learn version. Following is the log that I am encountering. Any pointers from you will be really appreciated. 
```
(ccis) iacvlab@iacvlab-desktop:~/anurag/ccis/dynamic-re-ranking$ ./run.sh 
------------------------------------------------------------
cfg.__dict__
{'base_lr': 0.0002,
 'ckpt_file': '/home/iacvlab/anurag/ccis/results/ckpt.pth',
 'crop_prob': 0,
 'crop_ratio': 1,
 'dataset': 'market1501',
 'epochs_per_val': 10000000000.0,
 'exp_decay_at_epoch': 151,
 'exp_dir': '/home/iacvlab/anurag/ccis/results/',
 'ids_per_batch': 32,
 'im_mean': [0.486, 0.459, 0.408],
 'im_std': [0.229, 0.224, 0.225],
 'ims_per_id': 4,
 'last_conv_stride': 1,
 'log_to_file': True,
 'lr_decay_type': 'exp',
 'margin': 0.3,
 'model_weight_file': '/home/iacvlab/anurag/ccis/weights/market1501_stride1/model_weight.pth',
 'normalize_feature': False,
 'only_test': True,
 'prefetch_threads': 2,
 'resize_h_w': (256, 128),
 'resume': False,
 'run': 1,
 'scale_im': True,
 'seed': None,
 'staircase_decay_at_epochs': (101, 201),
 'staircase_decay_multiply_factor': 0.1,
 'stderr_file': '/home/iacvlab/anurag/ccis/results/stderr_2019-11-18_14:47:07.txt',
 'stdout_file': '/home/iacvlab/anurag/ccis/results/stdout_2019-11-18_14:47:07.txt',
 'steps_per_log': 20,
 'sys_device_ids': (0,),
 'test_batch_size': 32,
 'test_final_batch': True,
 'test_mirror_type': None,
 'test_set_kwargs': {'batch_dims': 'NCHW',
                     'batch_size': 32,
                     'final_batch': True,
                     'im_mean': [0.486, 0.459, 0.408],
                     'im_std': [0.229, 0.224, 0.225],
                     'mirror_type': None,
                     'name': 'market1501',
                     'num_prefetch_threads': 2,
                     'part': 'test',
                     'prng': <module 'numpy.random' from '/home/iacvlab/anaconda3/envs/ccis/lib/python2.7/site-packages/numpy/random/__init__.pyc'>,
                     'resize_h_w': (256, 128),
                     'scale': True,
                     'shuffle': False},
 'test_shuffle': False,
 'total_epochs': 300,
 'train_final_batch': False,
 'train_mirror_type': 'random',
 'train_set_kwargs': {'batch_dims': 'NCHW',
                      'crop_prob': 0,
                      'crop_ratio': 1,
                      'final_batch': False,
                      'ids_per_batch': 32,
                      'im_mean': [0.486, 0.459, 0.408],
                      'im_std': [0.229, 0.224, 0.225],
                      'ims_per_id': 4,
                      'mirror_type': 'random',
                      'name': 'market1501',
                      'num_prefetch_threads': 2,
                      'part': 'trainval',
                      'prng': <module 'numpy.random' from '/home/iacvlab/anaconda3/envs/ccis/lib/python2.7/site-packages/numpy/random/__init__.pyc'>,
                      'resize_h_w': (256, 128),
                      'scale': True,
                      'shuffle': True},
 'train_shuffle': True,
 'trainset_part': 'trainval',
 'val_set_kwargs': {'batch_dims': 'NCHW',
                    'batch_size': 32,
                    'final_batch': True,
                    'im_mean': [0.486, 0.459, 0.408],
                    'im_std': [0.229, 0.224, 0.225],
                    'mirror_type': None,
                    'name': 'market1501',
                    'num_prefetch_threads': 2,
                    'part': 'val',
                    'prng': <module 'numpy.random' from '/home/iacvlab/anaconda3/envs/ccis/lib/python2.7/site-packages/numpy/random/__init__.pyc'>,
                    'resize_h_w': (256, 128),
                    'scale': True,
                    'shuffle': False},
 'weight_decay': 0.0005}
------------------------------------------------------------
----------------------------------------
market1501 test set
----------------------------------------
NO. Images: 31969
NO. IDs: 751
NO. Query Images: 3368
NO. Gallery Images: 15913
NO. Multi-query Images: 12688
----------------------------------------
Loaded model weights from /home/iacvlab/anurag/ccis/weights/market1501_stride1/model_weight.pth

=========> Test on dataset: market1501 <=========

Extracting feature...
1000/1000 batches done, +1.84s, total 95.58s
Done, 95.69s
Computing distance...
Done, 2.10s
Computing scores...
Done, 12.09s
Single Query:                 [mAP: 2.08%], [cmc1: 8.11%], [cmc5: 17.10%], [cmc10: 23.16%]
Multi Query, Computing distance...
Done, 2.09s
Multi Query, Computing scores...
Done, 12.09s
Multi Query:                  [mAP: 2.93%], [cmc1: 11.82%], [cmc5: 24.76%], [cmc10: 31.21%]
Re-ranking based on proposed algo...
```

**Comment by Anurag14** [comment]
Thanks for your time. I was initially running code in python3 environment. After changing the environment back to exactly like described in readme code started behaving absurdly. (poor results)

I cloned repo again and it fixed it.
